README.md

```diff
@@ -128,23 +128,49 @@ with:
 
 ### Team access
 
+Create or update existing team access resources. `team_id` takes precedence over `team_name` if both are supplied. Expressions can be used to reference remote state attributes.
+
 ```yml
 with:
   team_access: |-
     - team_name: Readers
       access: read
-    - team_name: Custom
+    - team_id: team-12345
+      access: write
+    - team_name: ${data.terraform_remote_state.tfe.outputs.teams["Engineering"].name}
       permissions:
         runs: read
         variables: read
         state_versions: read
         sentinel_mocks: read
         workspace_locking: true
+  remote_states: |-
+    tfe:
+      backend: remote
+      config:
+        bucket: s3-bucket
+        key: terraform.tfstate
+        region: us-east-1
+```
+
+To import existing team access resources, a static value for `team_id` must be supplied
+
+```yml
+with:
+  import: true
+  team_access: |-
+    - team_name: Readers
```

Comment:
Why allow both `team_name` and `team_id`? I suggested splitting with the thought that it was necessary for disambiguating IDs from names, but TF will resolve any expressions and `^team-` will identify team IDs.

If a name _and_ an ID are supplied but they don't agree, the correct behavior would be indeterminate. The only safe thing to do is to error and exit.

You could support both for clarity, but I think just `team` that is either an ID or a name should work fine.

Reply:
Yeah i can poke at that. My thought was that making the documentation clear that ID takes precedence over name would be a reasonable. 

In practice it felt like a nice experience to just throw the ID in there for import and then delete it without having to change and change back

Reply:
Avoid leaning too heavily on docs or coupling tools too tightly to your own workflow instincts. It is more important that software be correct (or predictable) than convenient, in the long run.

> In practice it felt like a nice experience to just throw the ID in there for import and then delete it without having to change and change back

A code comment handles this without creating the opportunity for ambiguous/invalid inputs.

Reply:
The prospect of importing by static name and switching to an ID expression in a future commit is awkward. Too cumbersome across 10s or 100s of repos. It's good that it's _possible_ but don't worry about making it _optimal_. Save that energy for making imports work without caveats.

Reply:
ah i knew there was something but I couldn't quite remember

If both name and ID input under the same key, I don't believe there's a way to tell whether an expression points to a team ID or a team name. This is important to determine whether to add a the team data block and use that as a reference or not.

How about we error if they're both supplied?

Reply:
> If both name and ID input under the same key, I don't believe there's a way to tell whether an expression points to a team ID or a team name. This is important to determine whether to add a the team data block and use that as a reference or not.

This was my original thought. But that's only true if you're thinking of the inputs as static strings, e.g., in Go. At the stage of runtime we're talking about, Terraform only cares if values are _known_. Any expression that is known can be evaluated and used in `for_each`. 

Team IDs have a prescribed format (`team-*`), that can be used to determine which inputs are team IDs versus names:

https://github.com/TakeScoop/terraform-cloud-workspace-action/pull/30#issuecomment-897252999

Anything that is not a team ID could be assumed to be a team name. I just checked though, and that assumption isn't strictly true. If you want, you can create a team named `team-foo`, or even copy the exact ID of another team as a name. So...

> How about we error if they're both supplied?

This seems like the best option. Turns out you really do need additional info to unambiguously identify the team after all.

Minor:

```yaml
team: # either or
  id: team-abc123
  name: Foo
```

Nesting versus prefixing guarantees that `name` and `id` will be next to each other so the source of the conflict is clear. Not particularly important given that team access objects have 4 possible top level keys. Would have a more appreciable impact if you had 20 keys and `team_id` and `team_name` could end up physically far from each other.

Reply:
I see, yes we could have pushed that is-team-id check within terraform instead of in Go, and let Terraform conditionally construct the tfe_team block and the team access team ID ref. 

Also good point regarding team names as IDs.

Reply:
Yep, anything where you're going to allow an expression, you cannot consume the value in Go, only pass it.